src/main/java/de/tum/in/www1/artemis/repository/ExerciseRepository.java

```diff
@@ -626,4 +629,13 @@ void revokeAccessToRestrictedFeedbackSuggestionModulesByCourseId(@Param("courseI
                 AND TYPE (e) IN (ModelingExercise, TextExercise, ProgrammingExercise)
             """)
     Set<Exercise> findAllExercisesWithPotentialPlagiarismByExamId(@Param("examId") long examId);
+
+    @Query("""
+            SELECT count(e) > 0
+            FROM Exercise e
+                LEFT JOIN e.exerciseGroup eg
+            WHERE eg IS NOT NULL
+                AND e.id = :exerciseId
+            """)
```

Comment:
Not sure if necessary but one could change the LEFT JOIN to an INNER JOIN to improve performance or use EXISTS instead of COUNT (as exists stops after finding a match)


Similar to:
```suggestion
    @Query("""
           SELECT EXISTS (
               SELECT 1
               FROM Exercise e
               WHERE e.exerciseGroup IS NOT NULL
               AND e.id = :exerciseId
           )
       """)
```

Reply:
The shortest version that agrees with the JPQL grammar would be 
```
SELECT CASE WHEN EXISTS(
                SELECT 1
                FROM Exercise e
                WHERE e.exerciseGroup IS NOT NULL
                    AND e.id = :exerciseId
            ) THEN TRUE ELSE FALSE END
```
Although the other query seems to work for now any minor updates could break the query.